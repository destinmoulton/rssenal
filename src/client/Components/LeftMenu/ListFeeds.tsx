import { OrderedMap } from "immutable";
import * as React from "react";
import { connect } from "react-redux";

import FeedItem from "./FeedItem";

import { IFeed, TFeedID, TFolderID, TFeeds } from "../../interfaces";

interface IListFeedsProps {
    feeds: TFeeds;
    editFeed: (feed: IFeed) => void;
    folderId: TFolderID;
}

const ListFeeds = (props: IListFeedsProps) => {
    const { editFeed, feeds, folderId } = props;

    const feedList = feeds
        .filter((feed: IFeed) => {
            return feed.folder_id === folderId;
        })
        .toArray()
        .map((feed: IFeed) => {
            return <FeedItem key={feed._id} feed={feed} editFeed={editFeed} />;
        });

    return (
        <div>
            {feedList}
            <div className="clear" />
        </div>
    );
};

export default ListFeeds;
