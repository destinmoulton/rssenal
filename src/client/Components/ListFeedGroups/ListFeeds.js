import PropTypes from "prop-types";
import React, { Component } from "react";
import { connect } from "react-redux";

class ListFeeds extends Component {
    static propTypes = {
        groupId: PropTypes.string.isRequired
    };

    render() {
        const { feeds, groupId } = this.props;

        const feedList = [];
        const groupFeeds = feeds.filter((feed)=>{
            return (feed.feedgroup_id === groupId);
        }).map((feed)=>{
            feedList.push(<div key={feed._id}>{feed.title}</div>)
        });
        
        return (
            <div>
                {feedList}
            </div>
        );
    }
}
const mapStateToProps = (state)=>{
    const { feeds } = state;

    return {
        feeds: feeds.feeds
    }
}

const mapDispatchToProps = (dispatch)=>{
    return {

    }
}
export default connect(mapStateToProps, mapDispatchToProps)(ListFeeds);