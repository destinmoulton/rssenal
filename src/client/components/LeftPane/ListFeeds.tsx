import debug from "debug";
import * as React from "react";

import FeedItemContainer from "../../containers/LeftPane/FeedItemContainer";

import { IFeed, TFolderID, TFeeds } from "../../types";

const log = debug("rssenal:ListFeeds");

interface IListFeedsProps {
    feeds: TFeeds;
    editFeed: (feed: IFeed) => void;
    folderId: TFolderID;
}

const ListFeeds = (props: IListFeedsProps) => {
    log("ListFeeds()");
    const { editFeed, feeds, folderId } = props;

    const feedList = feeds
        .filter((feed: IFeed) => {
            return feed.folder_id === folderId;
        })
        .toArray()
        .map((feed: IFeed) => {
            return (
                <FeedItemContainer
                    key={feed._id}
                    feed={feed}
                    editFeed={editFeed}
                />
            );
        });

    return (
        <div>
            {feedList}
            <div className="clear" />
        </div>
    );
};

export default ListFeeds;
