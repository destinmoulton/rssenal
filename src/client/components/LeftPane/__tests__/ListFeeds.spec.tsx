import * as React from "react";

import { shallow } from "enzyme";
import ListFeeds from "../ListFeeds";

import feeds from "../../../../../test/data/feeds";

describe("<ListFeeds />", () => {
    it("renders and matches the snapshot", () => {
        const wrapper = shallow(
            <ListFeeds
                feeds={feeds}
                editFeed={jest.fn()}
                folderId={"5a54f74838e78863ac6a8961"}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
