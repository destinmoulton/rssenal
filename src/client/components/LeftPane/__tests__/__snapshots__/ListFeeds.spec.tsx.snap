// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFeeds /> renders and matches the snapshot 1`] = `
<div>
  <div
    className="clear"
  />
</div>
`;
